test/thorest-api/logs/query-events.test.ts

```diff
@@ -1,39 +1,627 @@
+            ).toBeTrue()
+
+            const blockNumbers = response.body?.map(
+                (log) => log?.meta?.blockNumber,
+            )
+
+            expect(
+                blockNumbers,
+                'Should be an array of block numbers',
+            ).toBeArray()
+            expect(
+                blockNumbers,
+                'Should be sorted in the correct order',
+            ).toEqual(
+                order === 'asc' || order === undefined
+                    ? (blockNumbers as number[]).sort((a, b) => a - b)
+                    : (blockNumbers as number[]).sort((a, b) => b - a),
+            )
+        }
+
+        it('events should be ordered by DESC', async () => {
+            await runQueryEventLogsTest('desc')
+        })
+
+        it('events should be ordered by ASC', async () => {
+            await runQueryEventLogsTest('asc')
+        })
+
+        it('default should be asc', async () => {
+            await runQueryEventLogsTest(undefined)
+        })
+    })
+
+    describe('query by "options"', () => {
+        it('should be able omit all the options', async () => {
+            const transfer = readRandomTransfer()
+
+            const baseRequest = buildRequestFromTransfer(transfer)
+            const request = {
+                ...baseRequest,
+                options: null,
+            }
+
+            const eventLogs = await Node1Client.queryEventLogs(request)
+
+            expect(
+                eventLogs.success,
+                'API response should be a success',
+            ).toBeTrue()
+            expect(eventLogs.httpCode, 'Expected HTTP Code').toEqual(200)
+            expect(
+                eventLogs.body?.some((log) => log?.meta?.txID),
+                'The response should contain the relevant event log',
+            ).toBeTrue()
+        })
+
+        it('should be able to omit the "offset" field', async () => {
+            const transfer = readRandomTransfer()
+
+            const baseRequest = buildRequestFromTransfer(transfer)
+            const request = {
+                ...baseRequest,
+                options: {
+                    limit: 10_000,
+                    offset: undefined,
                 },
-            ],
+            }
+
+            const eventLogs = await Node1Client.queryEventLogs(request)
+
+            expect(
+                eventLogs.success,
+                'API response should be a success',
+            ).toBeTrue()
+            expect(eventLogs.httpCode, 'Expected HTTP Code').toEqual(200)
+            expect(
+                eventLogs.body?.some((log) => log?.meta?.txID),
+                'The response should contain the relevant event log',
+            ).toBeTrue()
         })
 
-        assert(eventLogs.success, 'eventLogs.success is false')
+        it('should be able paginate requests', async () => {
+            const { firstBlock, lastBlock } = transferDetails
+
+            const pages = 5
```

Comment:
are there limits for these values ? maybe we can check them too (like min and max for limit and offset)

Reply:
There are no max numbers AFAIK because there can be many millions of events. We can omit the limit and offset and query the whole chain